### PR TITLE
[568983] Project renaming fails for names containing non-alphanumerical

### PR DESCRIPTION
**Pull request by sandupostaru**
characters

- Fix resource reference renaming issue
- Fix file reference renaming issue in element description
- Update renaming documentation
- Add JUnit

Change-Id: Ie90f5d46a11b90f269dc1a12c8c991ac3782f43e
Signed-off-by: Sandu Postaru <sandu.postaru@thalesgroup.com>

### CODE REVIEW

File: core/plugins/org.polarsys.capella.core.platform.sirius.ui.navigator/src/org/polarsys/capella/core/platform/sirius/ui/navigator/actions/RenameResourceAction.java [hunk 1]
```diff
@@ -792,31 +804,98 @@ private void updateReferencesFor(IResource[] members_p, String newName_p, String
...
-    // Handle special case.
-    if (oldName.contains(Character.toString('&'))) {
-      oldName = replaceSpecialCharacter(oldName, '&', AND_HTML_REPRESENTATION);
-      //Replace the entire string not in a subString
-      newContent = newContent.replaceAll("\\b" + Pattern.quote(oldName) + "\\b", newName);
+      return content.replaceAll(regex, replacement);
     }
 
-    return newContent;
+    return content;
```
Comment (sandupostaru) [inline core/plugins/org.polarsys.capella.core.platform.sirius.ui.navigator/src/org/polarsys/capella/core/platform/sirius/ui/navigator/actions/RenameResourceAction.java:898]: This is the core change

File: core/plugins/org.polarsys.capella.core.platform.sirius.ui.navigator/src/org/polarsys/capella/core/platform/sirius/ui/navigator/actions/RenameResourceAction.java [hunk 2]
```diff
@@ -792,31 +804,98 @@ private void updateReferencesFor(IResource[] members_p, String newName_p, String
...
+   */
+  private String updateFileContent(String initialFileContent, String oldName, String newName) {
+    String updatedFileContent = replaceRaw(initialFileContent, oldName, newName);
+
+    updatedFileContent = replaceEncoded(updatedFileContent, oldName, newName);
+
+    updatedFileContent = replaceLegacy(updatedFileContent, oldName, newName);
+
+    return updatedFileContent;
+  }
```
Comment (sandupostaru) [inline core/plugins/org.polarsys.capella.core.platform.sirius.ui.navigator/src/org/polarsys/capella/core/platform/sirius/ui/navigator/actions/RenameResourceAction.java:824]: This is the core change

File: tests/plugins/org.polarsys.capella.test.model.ju/src/org/polarsys/capella/test/model/ju/rename/RenameSpecialCharTestCase.java [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+    assertFalse(fileNotRenamedMsg, projectFile.getName().equals(renameData.initialName));
+  }
+
+  private void assertValidSession(IFile renamedFile) {
+    IFile airdFile = getAirdFile(renamedFile);
+    URI airdFileUri = EcoreUtil2.getURI(airdFile);
+    Session session = SessionManager.INSTANCE.getSession(airdFileUri, new NullProgressMonitor());
+
+    assertNotNull(MessageFormat.format("Invalid Session for renamed file {0}", renamedFile.getName()), session);
+    session.close(new NullProgressMonitor());
```
Comment (pdulth) [inline tests/plugins/org.polarsys.capella.test.model.ju/src/org/polarsys/capella/test/model/ju/rename/RenameSpecialCharTestCase.java:130]: GuiActions.closeSession

File: tests/plugins/org.polarsys.capella.test.model.ju/src/org/polarsys/capella/test/model/ju/rename/RenameSpecialCharTestCase.java [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+  @Override
+  public List<String> getRequiredTestModels() {
+    return Arrays.asList(TEST_MODEL_NAME);
+  }
+
+  @Override
+  public void test() throws Exception {
+
+    List<RenameData> firstRenameData = Arrays.asList( //
+        new RenameData("[rename model].capella", "superman.capella"), //
```
Comment (pdulth) [inline tests/plugins/org.polarsys.capella.test.model.ju/src/org/polarsys/capella/test/model/ju/rename/RenameSpecialCharTestCase.java:63]: copyrighted names i guess

File: tests/plugins/org.polarsys.capella.test.model.ju/src/org/polarsys/capella/test/model/ju/rename/RenameSpecialCharTestCase.java (hunk 3 above)
Comment (sandupostaru) [reply tests/plugins/org.polarsys.capella.test.model.ju/src/org/polarsys/capella/test/model/ju/rename/RenameSpecialCharTestCase.java]: Done

File: tests/plugins/org.polarsys.capella.test.model.ju/src/org/polarsys/capella/test/model/ju/rename/RenameSpecialCharTestCase.java (hunk 4 above)
Comment (sandupostaru) [reply tests/plugins/org.polarsys.capella.test.model.ju/src/org/polarsys/capella/test/model/ju/rename/RenameSpecialCharTestCase.java]: Done

**Review by pdulth — APPROVED** [review]

**Pull request merged**